src/App.php

```diff
@@ -297,16 +299,236 @@ public function __construct($basepath)
 
 		// Register template engines
 		$this->register_template_engine('Friendica\Render\FriendicaSmartyEngine');
+	}
+
+	/**
+	 * Load the configuration files
+	 *
+	 * First loads the default value for all the configuration keys, then the legacy configuration files, then the
+	 * expected local.ini.php
+	 */
+	private function loadConfigFiles()
+	{
+		$this->loadConfigFile($this->basepath . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR . 'defaults.ini.php');
 
-		/**
-		 * Load the configuration file which contains our DB credentials.
-		 * Ignore errors. If the file doesn't exist or is empty, we are running in
-		 * installation mode.	 *
+		// Legacy .htconfig.php support
+		if (file_exists($this->basepath . DIRECTORY_SEPARATOR . '.htpreconfig.php')) {
+			$a = $this;
+			include $this->basepath . DIRECTORY_SEPARATOR . '.htpreconfig.php';
+		}
+
+		// Legacy .htconfig.php support
+		if (file_exists($this->basepath . DIRECTORY_SEPARATOR . '.htconfig.php')) {
+			$a = $this;
+
+			include $this->basepath . DIRECTORY_SEPARATOR . '.htconfig.php';
+			unset($db_host, $db_user, $db_pass, $db_data);
+
+			if (isset($default_timezone)) {
+				$this->setConfigValue('system', 'default_timezone', $default_timezone);
+				unset($default_timezone);
+			}
+
+			if (isset($pidfile)) {
+				$this->setConfigValue('system', 'pidfile', $pidfile);
+				unset($pidfile);
+			}
+		}
+
+		if (file_exists($this->basepath . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR . 'local.ini.php')) {
+			$this->loadConfigFile($this->basepath . DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR . 'local.ini.php');
+		}
+	}
+
+	/**
+	 * Tries to load the specified configuration file into the App->config array.
+	 * Overwrites previously set values.
+	 *
+	 * The config format is INI and the template for configuration files is the following:
+	 *
+	 * <?php return <<<INI
+	 *
+	 * [section]
+	 * key = value
+	 *
+	 * INI;
+	 * // Keep this line
+	 *
+	 * @param type $filepath
+	 * @throws Exception
+	 */
+	public function loadConfigFile($filepath)
+	{
+		if (!file_exists($filepath)) {
+			throw new Exception('Error parsing non-existent config file ' . $filepath);
+		}
+
+		$contents = include($filepath);
```

Comment:
Out of curiosity, what is the benefit of using this instead of `parse_ini_file()`?

Reply:
To prevent direct display of the INI file in the browser since the `config` folder is under the web root.